### ISSUE TITLE
Neural Network

### ISSUE THREAD

**Issue by Janecxl** [Janecxl/Notes#6]
## **What is Neural Network 🤔**
**Neural Network** is a function powered by a series of inputs X, and weights W, that run through activation function F and produce an output y.

## **Pros and Cons**
✅ Good at modeling non-linear data
❌ Hard to understand intuitively
❌ Computationally expensive

## **Forward Propagation and Backward Propagation**
**Forward propagation** is how neural networks make predictions. Input data is “forward propagated” through the network layer by layer to the final layer which outputs a prediction. 

**Backward propagation** adjusts weights based on the error rate obtained in the previous epoch. Reduce error iteratively.

## **Perceptron** 
**Perceptron** is a classifier that produces binary outputs based on the weighted sum of the inputs. It is the foundational building block of neural networks. 
![](https://latex.codecogs.com/svg.latex?&space;F(x_1,...x_n,w_1,...w_n)=1\text{if}\sum\omega_ix_i\geq\theta\text{else}0)

## **Reiterate learning rules**
**Perceptron Rule** guarantee finite convergency if the problem is linearly separable.
![](https://latex.codecogs.com/svg.latex?&space;\Delta\omega_i=\eta(y-\hat{y})x_i) 
![](https://latex.codecogs.com/svg.latex?&space;\eta) is learning rate. 

**Gradient Descent** uses calculus. It is robust, converges to the local optimum.
![](https://latex.codecogs.com/svg.latex?&space;\Delta\omega_i=\eta(y-a)x_i) 
![](https://latex.codecogs.com/svg.latex?&space;a=\sum_ix_i\omega_i) 

#### **Activation function**
**Sigmoids** 
![](https://latex.codecogs.com/svg.latex?&space;\sigma(a)=\frac{1}{1+e^{-a}}) 

#### **Learning Rate**
**Learning Rate** controls adjustments so that we take small steps in the right direction rather than overshooting.
A low learning rate slows down the learning process but converges smoothly. A larger learning rate speeds up the learning but may not converge.

#### **Hidden Layer**
The number of hidden layers and the number of neurons in each hidden layer.

## **Loss function**
**Cross-Entropy**
Cross-entropy is a measure of the difference between two probability distributions for a given random variable. Cross-entropy loss increases as the predicted probability diverge from the actual label.

## **Reference 📖**
[Machine Learning Glossary](https://ml-cheatsheet.readthedocs.io/en/latest/)
[Wikipedia Neural network](https://en.wikipedia.org/wiki/Neural_network)
[Scikit-learn Neural network](https://scikit-learn.org/stable/modules/neural_networks_supervised.html)
[Cross Entropy](https://machinelearningmastery.com/cross-entropy-for-machine-learning/)
Gatech CS7641: [Machine Learning notes](https://teapowered.dev/assets/ml-notes.pdf)
